### DISCUSSION TITLE
Swap Program question

### DISCUSSION THREAD

**Discussion by OrionRiftStudios** [bmresearch/Solnet#295] [Q&A] [answered]
Hello,

I am building a dapp where I would like to offer the possibility to swap tokens like panckeswap( basic version)

In general I understand the concepts for swap but I when I was studying the examples in the repo it was confusing for me.

I would like to know if someone knows where I could find more information about this topic?. 

Thank you

**Comment by hoakbuilds** [comment]
@thesoftwarejedi Might be the most knowledgeable person about this out of all the contributors in the SDK.

**Comment by thesoftwarejedi** [comment, marked as answer]
Thanks @murlokito Happy to answer any questions.  I wrote the swap integration for Solnet, still pending a commit of a step finance swap integration.  Would love to do a fancy jupiter swap port - but thatd be a lot of work.  however, it'd cover ALL swaps.  Lot's of client side fanciness in jupiter to toy with txs.

All swap programs are slightly different - each platform on solana appears to run a modified fork of tokenswap.

I wrote the TokenSwap onchain program for Step Finance (well, I forked tokenswap and mode a handful of modifications)

What questions can I help you with @OrionRiftStudios

**Reply by OrionRiftStudios** [reply]
Thank you!
